src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionContext.FeatureCollection.cs

```diff
@@ -11,25 +14,38 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.Internal
 {
     internal sealed partial class QuicConnectionContext : IProtocolErrorCodeFeature, ITlsConnectionFeature
     {
+        private X509Certificate2? _clientCert;
+        private Task<X509Certificate2?>? _clientCertTask;
+
         public long Error { get; set; }
 
         // Support accessing client certificate
         // https://github.com/dotnet/aspnetcore/issues/34756
         public X509Certificate2? ClientCertificate
         {
-            get => throw new NotSupportedException();
-            set => throw new NotSupportedException();
+            get { return _clientCert ??= ConvertToX509Certificate2(_connection.RemoteCertificate); }
+            set { _clientCert = value; }
```

Comment:
Should this also set the `_clientCertTask`?
```suggestion
            set 
            {
                 _clientCert = value; 
                 _clientCertTask = Task.FromResult(value);
             }
```

Reply:
It should at least clear the _clientCertTask so GetClientCertificateAsync sees the new cert.